pkg/controller/petset/pet.go

```diff
@@ -81,6 +81,15 @@ type petSyncer struct {
 	blockingPet *pcb
 }
 
+// errUnhealthyPet is returned when a we either know for sure a pet is unhealthy,
+// or don't know its state but assume it is unhealthy. It's used as a signal to the caller for further operations like updating status.replicas.
+// This is not a fatal error.
+type errUnhealthyPet string
```

Comment:
The error doesn't contain the names of objects that caused the error. Also, I prefer to use type switch for error type checking instead of string comparison. Thanks for the suggestion, but I don't think this is something I would  like to implement in this change.